screeps-game-api/src/memory.rs

```diff
@@ -30,37 +83,79 @@ impl MemoryReference {
         MemoryReference(reference)
     }
 
-    pub fn bool(&self, path: &str) -> bool {
-        js_unwrap!(Boolean(@{self.as_ref()}[@{path}]))
+    pub fn bool(&self, key: &str) -> bool {
+        js_unwrap!(Boolean(@{self.as_ref()}[@{key}]))
+    }
+
+    pub fn path_bool(&self, path: &str) -> bool {
+        js_unwrap!(Boolean(_.get(@{self.as_ref()}, @{path})))
+    }
+
+    pub fn num(&self, key: &str) -> Option<f64> {
+        (js! {
+            return (@{self.as_ref()})[@{key}];
+        }).try_into()
+            .map(Some)
+            .unwrap_or_default()
     }
 
-    pub fn num(&self, path: &str) -> Option<f64> {
+    pub fn path_num(&self, path: &str) -> Option<f64> {
         (js! {
-            return (@{self.as_ref()})[@{path}];
+            return _.get(@{self.as_ref()}, @{path});
         }).try_into()
             .map(Some)
             .unwrap_or_default()
     }
 
-    pub fn int(&self, path: &str) -> Option<i32> {
+    pub fn int(&self, key: &str) -> Option<i32> {
         (js! {
-            return (@{self.as_ref()})[@{path}];
+            return (@{self.as_ref()})[@{key}];
         }).try_into()
             .map(Some)
             .unwrap_or_default()
     }
 
-    pub fn string(&self, path: &str) -> Option<String> {
+    pub fn path_int(&self, path: &str) -> Option<i32> {
         (js! {
-            return (@{self.as_ref()})[@{path}];
+            return _.get(@{self.as_ref()}, @{path});
```

Comment:
This probably deserves some discussion on what the best method is.

I originally used `= undefined` since it's strictly faster (or at least was in the V8 version that was benchmarked a while ago), but it's true that it doesn't exactly represent the right semantics.

Reply:
I though it would just store an empty object, but it turns out that it does get rid of it! Maybe a result of the serialization process.

In that case, I can revert the last commit!